Introduce JS output-type + collections (#41)
PicnicSupermarket/localicious: README.md

## The Localicipe

The central concept of localicious is the so-called Localicipe. It is a YAML file that contains all localized copy and accessibility strings grouped by feature and screen. The strings in the Localicipe can be divided into different collections. Multiple collections can be combined when [Converting the Localicipe](#converting-the-localicipe) into platform specific outputs.

Using collections it's easy to keep track of strings that are used on a single platform and strings that are shared across multiple platforms. 

> For an existing iOS and Android app, it could be usefull to create three different collections: 

Reviewer: ```suggestion
For an existing iOS and Android app, it could be useful to create three different collections: 
```

A similar file with the Dutch translations will be created as well if we request localicious to do so:

`localicious render ./copy.yaml ./output_path --outputTypes android --collections ANDROID,SHARED --languages en,nl`

> By changing the destinatiooutput type, like so:

Reviewer 2: ```suggestion
By changing the destination output type, like so:
```

> `localicious render ./copy.yaml ./output_path --outputTypes ios --collections IOS,SHARED --languages en`

Author: After writing [this](https://github.com/PicnicSupermarket/localicious/pull/41/files#r458894429), I discovered that I've used inconsistent casing here as well... 
I'll leave it until we decide which way to go :)

Reviewer 3: Also, we assume collection names with all caps.

`./bin/localicious render ./copy.yaml ./output_path --outputTypes android,ios --collections ANDROID,Shared --languages en,nl`

With a collection named `Shared`
```
Shared:
  Delivery:
    Widget:
```

Is not supported.

```
❌ Something went wrong rendering.
❌ Your Localicipe contains validation issues:
    -  should have required property 'SHARED'
```

But I think the error message is clear enough. I would not change

## Validating

Whenever we make changes to the Localicipe, it is important to verify that the format of the file is still correct.
Using the `validate` command, a Localicipe can be validated.

**Syntax**

`localicious validate <localicipe path> <output path>`

**Options**

`--collections/-c` (required)

> - The collections, defined in the Localicipe, that should be included into the output.

Reviewer: ```suggestion
- The collections, defined in the Localicipe, that should be validated.
```
